docs/features/network-security-controls/admin-network-policy.md

```diff
@@ -1,303 +1,1351 @@
+  (a tenant cannot receive traffic from any other tenant) as the cluster’s default
+  network security standard, thus delegating the responsibility of explicitly allowing
+  traffic from other tenants to the tenant owner using NetworkPolicies.
 
-NetworkPolicy API was designed mainly for namespace owners or application developers. They are thus namespace scoped. NetworkPolicy API is not suitable for administrators/network-administrators/security-operators of the cluster because of two main reasons:
+AdminNetworkPolicy has been designed precisely to solve these problems. It provides
+a comprehensive cluster-wide network security solution for cluster administrators using
+two cluster-scoped CRDs:
 
-* They are not cluster-scoped and hence its hard to define a network of policies that spawns across namespaces
-* They cannot be created before the namespace is created (kapi server expects the namespace to be created first); thus they cannot satisfy the requirements where admins may want policies in the cluster to be in place before the workloads are even created.
-* The design of NetworkPolicy API is implicit which means the deny is already set in place when we create a policy and then we are expected to do an allowList of rules in the policy. Network administrators prefer to have the power of defining what exactly to deny and allow instead of this implicit model.
+* AdminNetworkPolicy API and
+* BaselineAdminNetworkPolicy API
 
-Sample API:
+which are shipped by the [sig-network-policy-api](https://network-policy-api.sigs.k8s.io/)
+working group as out-of-tree Kubernetes APIs. Both these APIs can be used to define
+cluster-wide admin policies in order to secure your cluster’s network traffic. These APIs
+are currently in the v1alpha1 version and subject to change based on upstream developments
+before becoming stable v1.
 
+## Motivation
+
+NetworkPolicies in Kubernetes were created for developers to be able to secure their
+applications. Hence they are under the control of potentially untrusted application authors.
+They are namespace scoped and are not suitable for solving the above user stories for cluster
+administrators because:
+
+* They are not cluster-scoped and hence its hard to define a network of policies
+  that spans across namespaces - you'd have to duplicate the same policy over and over
+  in each namespace
+* They cannot be created before the namespace is created (KAPI server expects the namespace
+  to be created first since it's namespace scoped); thus they cannot satisfy the requirements
+  where admins may want the policies to be applicable to arbitrary future
+  namespaces that could get created in the cluster.
+
+Another motivating was to fix some of the usability aspects of the NetworkPolicy API.
+The design of NetworkPolicy API is implicit which means the deny is already set in
+place when we create a policy and then we are expected to do an allowList of rules
```

Comment:
This is a bit confusing to read, suggestion:
NetworkPolicy API is designed with an implicit deny-all policy, meaning that with the first created policy all traffic not allowed by it will be denied. NetworkPolicy only defines which connection will be allowed.

Reply:
yeah good point, its amazing what things in my head become spaghetti when it comes out on paper. let me change this to your suggestion

Reply:
> its amazing what things in my head become spaghetti when it comes out on paper.

nah, NetworkPolicy is nearly impossible to describe in a way that is both accurate and comprehensible. :grimacing:

Reply:
fixed.